docs/ProfilingNonJavaApplications.md

```diff
@@ -0,0 +1,91 @@
+# Profiling Non-Java applications
+
+The scope of profiling non-java applications is limited to the case when profiler is controlled
+programmatically from the process being profiled and with LD_PRELOAD. It is worth noting that 
+[dynamic attach](https://github.com/async-profiler/async-profiler/blob/master/docs/OtherUseCases.md#launching-as-an-agent) 
+which is available for Java is not supported for non-Java profiling.
+
+
+## C API
+Similar to the
+[Java API](https://github.com/async-profiler/async-profiler/blob/master/docs/OtherUseCases.md#using-java-api),
+there is a C API for using inside native applications.
+```
+typedef const char* asprof_error_t;
+typedef void (*asprof_writer_t)(const char* buf, size_t size);
+
+// Should be called once prior to any other API functions
+DLLEXPORT void asprof_init();
+typedef void (*asprof_init_t)();
+
+// Returns an error message for the given error code or NULL if there is no error
+DLLEXPORT const char* asprof_error_str(asprof_error_t err);
+typedef const char* (*asprof_error_str_t)(asprof_error_t err);
+
+// Executes async-profiler command using output_callback as an optional sink
+// for the profiler output. Returning an error code or NULL on success.
+DLLEXPORT asprof_error_t asprof_execute(const char* command, asprof_writer_t output_callback);
+typedef asprof_error_t (*asprof_execute_t)(const char* command, asprof_writer_t output_callback);
+```
+To use it in a C/C++ application, include asprof.h. Below is an example usage showing how to use async-profiler command with the API. The :
+```
+void test_output_callback(const char* buffer, size_t size) {
+    fprintf(stderr, "%s\n", buffer);
+}
+
+int main() {
+    dlerror();
+    void* lib = dlopen(path/to/libasyncProfiler.so", RTLD_NOW);
+    if (lib == NULL) {
+        printf("%s\n", dlerror());
+        dlclose(lib);
```

Comment:
`dlclose(NULL)`?

Reply:
Addressed in [e6282c8](https://github.com/async-profiler/async-profiler/pull/1029/commits/e6282c83bc0aa6007df310fd5e638e9d4d5ed6f1)